Makefile.in

```diff
@@ -352,7 +353,8 @@ maintainerclean: distclean
 # implicit rules
 #
 .c.$(OBJEXT):
-	$(CC) -I. -I$(srcdir) -I$(srcdir)/main $(DEFS) $(ALL_CPPFLAGS) $(ALL_CFLAGS) -c $<
+	dir="$$(dirname $@)"; [ -d "$$dir" ] || $(MKDIR_P) "$$dir"
+	$(CC) -I. -I$(srcdir) -I$(srcdir)/main $(DEFS) $(ALL_CPPFLAGS) $(ALL_CFLAGS) -o $@ -c $<
```

Comment:
BTW, while I think it's good, there is a common saying suggesting some weird old compilers don't understand `-o` together with `-c`.  I doubt we care, as even Automake is trying and dropping this, but it might be worth mentioning.